### ISSUE TITLE
FakePlayer's "connection" field

### ISSUE THREAD

**Issue by stepsword** [MinecraftForge/MinecraftForge#7604]
Maybe it's just me who's experienced this but whenever a modder does something like this:

```
if (player instanceof ServerPlayerEntity) {
    ((ServerPlayerEntity) player).connection.sendPacket(packet);
}
```

They are risking a null pointer exception by not checking if the ServerPlayerEntity is actually a FakePlayer, which has the `connection` field set to null.

I have solved this in my own mod by creating a `SafeFakePlayer` which has the connection field set to a `FakePlayNetHandler extends ServerPlayNetHandler` where all the methods are overridden to do nothing. 

I feel like something like this should be part of the default FakePlayer, because the `.connection.sendPacket()` syntax is heavily used by vanilla and has the potential to cause lots of crashes if the wrong path gets executed with a FakePlayer.

**Comment by baileyholl** [comment]
I have run into this issue as well, this causes massive issues when fake players interact with blocks that may contain a GUI. Giving the fake player an empty ServerPlayNetHandler, or at minimum making OpenContainer in FakePlayer an empty method such that it doesn't cause a crash when it tries to invoke its own null connection would be fantastic.

**Comment by IzzelAliz** [comment]
There are chances that a real ServerPlayerEntity has null `connection`, for example when player joining server. The `PlayerEvent.LoadFromFile` event is called at this period, and possibly `PlayerWakeupEvent` since mc will [try to wake up player when loading player data](https://pastebin.com/Ps3uWKxY).
It would be great if any potential fix can give an empty connection to normal players too.

**EDIT** Can cause side effects so better not.

**Comment by TheCurle** [comment]
NFT: Closed by #7733.
